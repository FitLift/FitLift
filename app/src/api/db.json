{
  "currentExercises": [
    {
      "reps": 12,
      "time": 1517968200,
      "type": "Bicep Curls"
    },
    {
      "reps": 1,
      "time": 1517968200,
      "type": "Shoulder Press"
    },
    {
      "reps": 6,
      "time": 1517968200,
      "type": "Lateral Raises"
    }
  ]
}
