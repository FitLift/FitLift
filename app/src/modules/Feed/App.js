import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    flex: 1,
    justifyContent: 'center',
  },
  welcome: {
    fontSize: 20,
    margin: 10,
    textAlign: 'center',
  },
});

const Friends = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Feed
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Feed' })}
      title="Feed"
    />
  </View>
);

Friends.navigationOptions = {
  title: 'Feed',
};

export default Friends;
