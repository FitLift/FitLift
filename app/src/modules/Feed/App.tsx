import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Friends = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Feed
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Feed' })}
      title="Feed"
    />
  </View>
);

Friends.propTypes = {
  navigation: PropTypes.object.isRequired,
};

Friends.navigationOptions = {
  title: 'Feed',
};

export default Friends;
