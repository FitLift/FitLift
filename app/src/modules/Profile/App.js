import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Profile = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Profile
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Logout' })}
      title="Log Out"
    />
  </View>
);

Profile.propTypes = {
  navigation: PropTypes.object.isRequired,
};

Profile.navigationOptions = {
  title: 'Profile',
};

export default Profile;
