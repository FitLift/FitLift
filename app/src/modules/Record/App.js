import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const record = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      record
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'record' })}
      title="record"
    />
  </View>
);

record.propTypes = {
  navigation: PropTypes.object.isRequired,
};

record.navigationOptions = {
  title: 'record',
};

export default record;
