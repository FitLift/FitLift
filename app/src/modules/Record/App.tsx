import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Record = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Record
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Logout' })}
      title="Logout"
    />
  </View>
);

Record.propTypes = {
  navigation: PropTypes.object.isRequired,
};

Record.navigationOptions = {
  title: 'Record',
};

export default Record;
