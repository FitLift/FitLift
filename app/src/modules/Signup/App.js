import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Login = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.welcome}>
      login
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Signup' })}
      title="Signup"
    />
  </View>
);

Login.navigationOptions = {
  title: 'Sign up',
};

export default Login;
