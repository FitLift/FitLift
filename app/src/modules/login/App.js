import React from 'react';
import { Button, Text, View } from 'react-native';

const login = ({ navigation }) => (
  <View>
    <Text>login</Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'login' })}
      title="Log in"
    />
  </View>
);

login.navigationOptions = {
  title: 'Log In'
};

export default login;
