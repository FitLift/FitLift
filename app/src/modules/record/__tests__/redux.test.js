import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import reducer, {
  TypeKeys,
  fetchCurrentExercises,
} from '../redux';
import { mapStateToProps } from '../App';
import { API_URL } from '../../../../config';
import { currentExercises } from '../../../api/db.json';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('async action creator tests', () => {
  it('should fetch current exercises', () => {
    nock(API_URL)
      .get('/currentExercises/5')
      .reply(200, currentExercises);

    const store = mockStore();
    return store.dispatch(fetchCurrentExercises(5))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});

describe('reducer tests', () => {
  it('TypeKeys.LOGIN should work', () => {
    expect(reducer(
      {
        data: undefined,
        isLoading: false,
      },
      {
        type: TypeKeys.LOGIN,
      },
    )).toEqual({
      data: undefined,
      isLoading: true,
    });
  });

  it('TypeKeys.RECEIVE_CURRENT_EXERCISES should work', () => {
    expect(reducer(
      {
        data: undefined,
        isLoading: true,
      },
      {
        data: currentExercises,
        type: TypeKeys.RECEIVE_CURRENT_EXERCISES,
      },
    )).toEqual({
      data: currentExercises,
      isLoading: false,
    });
  });
});

describe('mapStateToProps tests', () => {
  it('should work', () => {
    expect(mapStateToProps({
      record: {
        data: undefined,
        isLoading: false,
      },
    })).toEqual({
      record: {
        data: undefined,
        isLoading: false,
      },
    });
  });
});
