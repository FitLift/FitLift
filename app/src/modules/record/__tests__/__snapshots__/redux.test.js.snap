// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selector tests it should work 1`] = `
Array [
  Object {
    "id": "-L5GDNmo8uZxet9S7mE2",
    "reps": 12,
    "timeStamp": 1518407604,
    "type": "Bicep Curls",
    "weight": 35,
  },
  Object {
    "id": "-L5GDeJnjbOj1SkQRJK8",
    "reps": 5,
    "timeStamp": 1518407604,
    "type": "Shoulder Press",
    "weight": 15,
  },
  Object {
    "id": "-L5GDeeBKCPlTnm-yCiX",
    "reps": 6,
    "timeStamp": 1518407604,
    "type": "Lateral Raises",
    "weight": 45,
  },
]
`;
